src/core/device.c

```diff
@@ -359,7 +359,7 @@ static int device_setup_unit(Manager *m, struct udev_device *dev, const char *pa
 fail:
         log_unit_warning_errno(u, r, "Failed to set up device unit: %m");
 
-        if (delete && u)
+        if (delete)
```

Comment:
@poettering , `mount_setup_unit` and `swap_setup_unit` contain `delete && u` too
```
$ git grep 'delete && u'
src/core/device.c:        if (delete && u)
src/core/mount.c:        if (delete && u)
src/core/swap.c:        if (delete && u)
```